packages/core/src/controllers/verify.ts

```diff
@@ -1,76 +1,130 @@
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+    if (!this.publicKey) {
+      await this.fetchAndPersistPublicKey();
     }
   };
 
   public register: IVerify["register"] = async (params) => {
-    if (!this.initialized) {
-      this.addToQueue(params.attestationId);
-      await this.init();
-    } else {
-      this.sendPost(params.attestationId);
+    if (!isBrowser()) return;
+    const { id, decryptedId } = params;
+    const url = `${this.verifyUrlV2}/attestation?projectId=${this.projectId}`;
+    let src = "";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ id, decryptedId }),
+      });
+      const { srcdoc } = await response.json();
+      src = srcdoc;
+      this.logger.debug("srcdoc fetched", src);
+    } catch (e) {
+      this.logger.warn(e);
+      return;
     }
+    try {
+      const document = getDocument() as Document;
+      const abortTimeout = this.startAbortTimer(ONE_SECOND * 3);
+      const attestatiatonJwt = await new Promise((resolve) => {
+        const abortListener = () => {
+          window.removeEventListener("message", listener);
+          document.body.removeChild(iframe);
+          throw new Error("attestation aborted");
+        };
+        this.abortController.signal.addEventListener("abort", abortListener, {
+          signal: this.abortController.signal,
+        });
+        const iframe = document.createElement("iframe");
+        iframe.srcdoc = src;
+        iframe.style.display = "none";
+        const listener = (event: MessageEvent) => {
+          if (!event.data) return;
+          const data = JSON.parse(event.data);
+          if (data.type === "verify_attestation") {
+            clearInterval(abortTimeout);
+            document.body.removeChild(iframe);
+            this.abortController.signal.removeEventListener("abort", abortListener);
+            window.removeEventListener("message", listener);
+            resolve(data.attestation === null ? "" : data.attestation);
+          }
+        };
+        document.body.appendChild(iframe);
+        window.addEventListener("message", listener, { signal: this.abortController.signal });
+      });
+      this.logger.debug("jwt attestation", attestatiatonJwt);
+      return attestatiatonJwt as string;
+    } catch (e) {
+      this.logger.warn(e);
+    }
+    return "";
   };
 
   public resolve: IVerify["resolve"] = async (params) => {
     if (this.isDevEnv) return "";
+    const { attestationId, hash } = params;
 
+    if (attestationId === "") {
+      this.logger.debug("resolve: attestationId is empty, skipping");
+      return;
+    }
+
+    if (attestationId) {
+      const validation = await this.isValidJwtAttestation(attestationId);
+      if (validation) return validation;
+    }
+    if (!hash) return;
```

Comment:
nope, because `hash` is used for verify v1, while verify v2 uses the `attestation` param. Since we first try with v2, `hash` is used only if `attestation` isn't available

Reply:
Got it, thanks for the context. Would be nice to have a comment about this.